iosxr_test.py

```diff
@@ -63,34 +66,41 @@ def check_result(result, success_message):
 
 def bringup_vagrant():
     '''
-    Bring up a vagrant box.
-
-    Clean up ssh keys from the generation of the virtualbox.
-
-    Use pxssh to fascilitate the ssh process.
+    Bring up a vagrant box and test the ssh connection.
     '''
+
     # Clean up Vagrantfile
     try:
         os.remove('Vagrantfile')
     except OSError:
         pass
 
+    global iosxr_port
+    global linux_port
+
+    # Use vagrant to init, add and bring up the inputted Vagrant VirtualBox
     logger.debug("Bringing up '%s'..." % input_box)
 
-    run(['vagrant', 'init', 'XRv64-test'])  # Single node for now, in future could bring up two nodes and do more testing
-    run(['vagrant', 'box', 'add', '--name', 'XRv64-test', input_box, '--force'])
+    output = run(['vagrant', 'init', 'XRv64-test'])
+    logger.debug(output)
```

Comment:
For the first, I did spend quite a bit of time trying to get the output to stream but the implmentation was never quite right - at least for a run() function that can be used in many ways. So I just went with the buffer, you're correct it's a bit slow but as long as you get the output I am fine - because you really only care of something goes wrong.

For the second, the run() command should catch errors when they occur and display the output. That part seems to work well.